src/Compilers/Core/Portable/Operations/IAssignmentExpression.cs

```diff
@@ -22,5 +22,16 @@ public interface IAssignmentExpression : IOperation
         /// </summary>
         IOperation Value { get; }
     }
+
+    /// <summary>
+    /// Represents an assignment expression.
+    /// </summary>
+    /// <remarks>
+    /// This interface is reserved for implementation by its associated APIs. We reserve the right to
+    /// change it in the future.
+    /// </remarks>
+    public interface IAssignmentExpression : IBaseAssignmentExpression
```

Comment:
What code will be consuming or implementing `IAssignmentExpression` rather than `IBaseAssignmentExpression`?